api/src/utils/transaction.ts

```diff
@@ -45,7 +62,7 @@ export const transaction = async <T = unknown>(knex: Knex, handler: (knex: Knex)
 				try {
 					return await knex.transaction((trx) => handler(trx));
 				} catch (error: any) {
-					if (error?.code !== COCKROACH_RETRY_ERROR_CODE) throw error;
+					if (error?.code !== COCKROACH_RETRY_ERROR_CODE && error?.code !== SQLITE_BUSY_ERROR_CODE) throw error;
```

Comment:
I think this should be an `||` as we need to throw in both the sqlite and cockroach case of the error. Additionally, since this can now be executed for both sqlite and cockroach we should probably add an explicit client check here as well, even though it is highly unlikely that the error codes overlap.
```suggestion
					if (
						(client === 'cockroachdb' && error?.code !== COCKROACH_RETRY_ERROR_CODE) ||
						(client === 'sqlite' && error?.code !== SQLITE_BUSY_ERROR_CODE)
					)
						throw error;
```

Reply:
Did consider it but choose the easy way first, thanks for the change!